### PR TITLE
Sync up with lint workflow in `r-lib/actions`

### PR DESCRIPTION
**Pull request by IndrajeetPatil**

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -19,16 +15,15 @@ jobs:
...
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
           extra-packages: |
-            any::gh
-            any::purrr
-          needs: check
-
-      - name: Install package
-        run: R CMD INSTALL .
+            r-lib/lintr
```
Comment (IndrajeetPatil) [inline .github/workflows/lint.yaml:24]: This is the only deviation from what's in the standard [example workflow](https://github.com/r-lib/actions/blob/v2-branch/examples/lint.yaml) for linting. 

I think we should always be linting with the latest GitHub version of `{lintr}`.

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -1,12 +1,8 @@
 on:
   push:
-    branches:
-      - main
-      - master
+    branches: [main, master]
```
Comment (MichaelChirico) [inline .github/workflows/lint.yaml:3]: any reason to keep `master` here? just to keep perfectly synced with the upstream source?

Comment (IndrajeetPatil) [reply .github/workflows/lint.yaml:3]: Yes, no other reason. Can also remove it. Fine either way.

Comment (MichaelChirico) [reply .github/workflows/lint.yaml:3]: SGTM, thanks for clarifying

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**